### PR TITLE
maxima: 5.46.0 -> 5.47.0, add patch for CVE-2024-34490

### PR DESCRIPTION
**Pull request by risicle**
## Description of changes

https://nvd.nist.gov/vuln/detail/CVE-2024-34490

When tests are enabled, receive the same 7/18244 test failures as without the patch.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/science/math/maxima/default.nix [hunk 1]
```diff
@@ -79,6 +79,8 @@ stdenv.mkDerivation (finalAttrs: {
       url = "https://raw.githubusercontent.com/sagemath/sage/07d6c37d18811e2b377a9689790a7c5e24da16ba/build/pkgs/maxima/patches/undoing_true_false_printing_patch.patch";
       sha256 = "0fvi3rcjv6743sqsbgdzazy9jb6r1p1yq63zyj9fx42wd1hgf7yx";
     })
+
```
Comment (doronbehar) [inline pkgs/applications/science/math/maxima/default.nix:82]: Could you put please a link here to upstream's commit?

File: pkgs/applications/science/math/maxima/default.nix [hunk 2]
```diff
@@ -93,7 +95,7 @@ stdenv.mkDerivation (finalAttrs: {
   #
   # These failures don't look serious. It would be nice to fix them, but I
   # don't know how and probably won't have the time to find out.
-  doCheck = false;    # try to re-enable after next version update
+  doCheck = true;    # try to re-enable after next version update
```
Comment (doronbehar) [inline pkgs/applications/science/math/maxima/default.nix:98]: Is this why this is a draft? 1 out of 1 tests failed so it seems according to the build log...

**Review by doronbehar — COMMENTED** [review]
Changes look good overall.

File: pkgs/applications/science/math/maxima/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/applications/science/math/maxima/default.nix:82]: I've put one in the header of the patch so it's self-describing

File: pkgs/applications/science/math/maxima/default.nix (hunk 2 above)
Comment (risicle) [reply pkgs/applications/science/math/maxima/default.nix]: No I was just careless staging my hunks. Removed.

**Comment by risicle** [comment]
I can't get the `sage` tests to pass on my machine (timeout?)

**Comment by doronbehar** [comment]
Result of `nixpkgs-review pr 329007` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>cantor (libsForQt5.cantor ,plasma5Packages.cantor)</li>
    <li>labplot</li>
    <li>maxima</li>
    <li>maxima-ecl</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>wxmaxima</li>
  </ul>
</details>

**Comment by doronbehar** [comment]
Result of `nixpkgs-review pr 329007` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>cantor (libsForQt5.cantor ,plasma5Packages.cantor)</li>
    <li>labplot</li>
    <li>maxima</li>
    <li>maxima-ecl</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>wxmaxima</li>
  </ul>
</details>

**Review by doronbehar — APPROVED** [review]

**Pull request merged**